File: benchmark/vector_dot_product/dot.js
function main() {
  const count = 6000000;
  const a = new BigInt64Array(count);
  const b = new BigInt64Array(count);

  for (let i = 0; i < count; i++) {
    a[i] = BigInt((i % 97) - 48);
    b[i] = BigInt((i % 89) - 44);
  }

  let sum = 0n;
  for (let i = 0; i < count; i++) {
    sum += a[i] * b[i];
  }

  console.log(`Vector dot: ${sum}`);
}

main();

File: benchmark/vector_dot_product/dot.c
#include <stdint.h>
#include <stdio.h>
#include <stdlib.h>

int main() {
    int count = 6000000;
    int64_t* a = (int64_t*)malloc((size_t)count * sizeof(int64_t));
    int64_t* b = (int64_t*)malloc((size_t)count * sizeof(int64_t));

    for (int i = 0; i < count; i++) {
        a[i] = (int64_t)((i % 97) - 48);
        b[i] = (int64_t)((i % 89) - 44);
    }

    int64_t sum = 0;
    for (int i = 0; i < count; i++) {
        sum += a[i] * b[i];
    }

    printf("Vector dot: %lld\n", (long long)sum);
    free(a);
    free(b);
    return 0;
}
